plugins/search/src/components/SearchResult/SearchResult.tsx

```diff
@@ -0,0 +1,230 @@
+      />
+      <Divider className={classes.divider} orientation="vertical" />
+      <Grid item sm={12}>
+        {searchTerm ? (
+          <Typography variant="h6">
+            {`${numberOfResults} `}
+            {numberOfResults > 1 ? `results for ` : `result for `}
+            <span className={classes.searchTerm}>"{searchTerm}"</span>{' '}
+          </Typography>
+        ) : (
+          <Typography variant="h6">{`${numberOfResults} results`}</Typography>
+        )}
+      </Grid>
+    </div>
+  );
+};
+
+const SearchResult = ({ currentTarget }: SearchResultProps) => {
+  const [showFilters, toggleFilters] = useState(false);
+  const [filters, setFilters] = useState<Filters>({
+    selected: 'All',
+    checked: [],
+  });
+
+  const [filteredResults, setFilteredResults] = useState<Array<object>>([]);
+
+  const searchApi = new SearchApi();
+
+  const { loading, error, value: results } = useAsync(() => {
+    return searchApi.getSearchData();
+  }, []);
+
+  useEffect(() => {
+    if (results) {
+      let withFilters = results;
+
+      // apply filters
+
+      // filter on selected
+      if (filters.selected !== 'All') {
+        withFilters = results.filter((result: any) =>
+          filters.selected.includes(result.kind),
+        );
+      }
+
+      // filter on checked
+      if (filters.checked.length > 0) {
+        withFilters = withFilters.filter((result: any) =>
+          filters.checked.includes(result.lifecycle),
+        );
+      }
+
+      // filter on searchTerm
+      if (currentTarget) {
+        withFilters = withFilters.filter(
+          (result: any) =>
+            result.name?.toLowerCase().includes(currentTarget) ||
+            result.description?.toLowerCase().includes(currentTarget),
+        );
+      }
+
+      setFilteredResults(withFilters);
+    }
+  }, [filters, currentTarget, results]);
+
+  if (loading || error || !results) return null;
+
+  const resetFilters = () => {
+    setFilters({
+      selected: 'All',
+      checked: [],
+    });
+  };
+
+  const updateSelected = (filter: string) => {
+    setFilters(prevState => ({
+      ...prevState,
+      selected: filter,
+    }));
+  };
+
+  const updateChecked = (filter: string) => {
+    if (filters.checked.includes(filter)) {
+      const index = filters.checked.indexOf(filter);
+      filters.checked.splice(index);
+
+      setFilters(prevState => ({
+        ...prevState,
+        checked: filters.checked,
+      }));
+      return;
+    }
+
+    setFilters(prevState => ({
+      ...prevState,
+      checked: [...prevState.checked, filter],
+    }));
+  };
+
+  return (
+    <>
+      <Grid container>
+        {showFilters && (
+          <Grid item sm={3}>
+            <Filters
+              filters={filters}
+              resetFilters={resetFilters}
+              updateSelected={updateSelected}
+              updateChecked={updateChecked}
+            />
+          </Grid>
+        )}
+        <Grid item sm={showFilters ? 9 : 12}>
```

Comment:
We should use `xs` right? i'm not sure i understand the grid system perfectly but it feels to me like if no `xs` is given it'll default to 12 and the view will jump to an unexpected state on very small screens. On the other hand ... a very small screen would have to have some kind of special treatment anyway because it's hard to fit stuff